### ISSUE TITLE
File not found error surfaces local path instead of cloud path

### ISSUE THREAD

**Issue by ejm714** [drivendataorg/cloudpathlib#118]
I was loading a file from s3 and mistyped the file location. The error I got showed me the local tmp path, which was misleading. I think this should be the standard error you get when you try to do a cloud operation on a file that doesn't exist (something like "head object does not exist") or at least something that includes the cloud path instead of the local path.

<img width="901" alt="Screen Shot 2021-02-05 at 3 33 22 PM" src="https://user-images.githubusercontent.com/22667367/107099905-c2d7af80-67c7-11eb-9224-966618d6d022.png">

**Comment by Mchristos** [comment]
I agree with this - it's really hard to debug the issue when I get this error. In my case, I'm passing a cloud path to another library, and the error happens inside it - so it's hard to know how to resolve the problem.

**Comment by pjbull** [comment]
Thanks for the comment @Mchristos, unfortunately our library just can't know what other libraries try to do with our objects so there's not simple way to inject ourselves into these error messages. This is really another instance of #128, which has a complicated set of tradeoffs for the possible approaches.

Right now we're leaning towards the experimental work in #322 as the best approach, so I'd be curious if that works in your case.

**Comment by pjbull** [comment]
Still an underlying issue, but can be worked around with patching built-ins:
https://cloudpathlib.drivendata.org/stable/patching_builtins/


```python
from cloudpathlib import patch_all_builtins

with patch_all_builtins():
    with open(cloud_path) as f:
        data = f.read()
```
